internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -381,6 +277,220 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 			return nil
 		}),
 	}
+
+	if features.FourPointOhBeta() {
```

Comment:
Could I suggest two things here specific to making breaking schema changes:

1. When we modify or rename an existing property we update the existing schema to reflect what the schema should look like going forward/post major release and use the inverted feature flag to overwrite/patch the schema with the old behaviour e.g.

```
resource := &pluginsdk.Resource{
...
Schema: map[string]*pluginsdk.Schema{
	"this_property_changes": {
		Type: pluginsdk.TypeString,
		Optional: true,
		Default: "is now this",
	},
}

if !features.FourPointOhBeta {
	resource.Schema["this_property_changes"] = {
		Type: pluginsdk.TypeString,
		Optional: true,
		Default: "was this",
	}	
}
```

This makes cleaning up post major release much easier, since all that's involved is to delete the entire block of code gated behind the feature flag.

2. For large blocks like this one where only one property within the block is changing, it's very helpful to limit the change to just that one property, what I mean is hopefully expressed better below

```
resource := &pluginsdk.Resource{
Read: dataSourceKubernetesClusterNodePoolRead,

Timeouts: &pluginsdk.ResourceTimeout{
	Read: pluginsdk.DefaultTimeout(5 * time.Minute),
},

Schema: map[string]*pluginsdk.Schema{
	"a_big_block": {
		Type:     pluginsdk.TypeList,
		Optional: true,
		Elem: &pluginsdk.Resource{
			Schema: map[string]*pluginsdk.Schema{
				"one_property_among_many": {
					Type:         pluginsdk.TypeString,
					Optional:     true,
					Default: "is now this",
				},
				... more properties down here
			},
		},
	},
}

if !features.FourPointOhBeta {
	resource.Schema["a_big_block"].Elem.(*pluginsdk.Resource).Schema["one_property_among_many"] = {
		Type: pluginsdk.TypeString,
		Optional: true,
		Default: "was this",
	}
}
```

This is much less code making the review of this change easier, but mostly later on when someone is putting together the upgrade guide with all the breaking changes, it's obvious at first glance what the breaking change is.

Reply:
Fixed.